### PR TITLE
Create a database based on a diagram

### PR DESCRIPTION
**Pull request by osmanwako**
### Inside this PR we did the following task:
- We create four tables based on the given diagrams
- We added the schema query on "schema_based_on_diagram.sql" file
- Join tables from "medical_histories"  and "treatments"  we have created a helper table called "medical_treatments"

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  );
+
+CREATE TABLE
+  medical_histories(
+    id INT GENERATED ALWAYS AS IDENTITY,
+    admitted_at date,
+    patient_id INT REFERENCES patients(id),
+    status VARCHAR(120),
+    PRIMARY KEY(id)
+  );
```
Comment (abdulhamiid) [inline schema_based_on_diagram.sql]: According to the database diagram, the column `admitted_at` datatype is `timestamp`

image

The `timestamp` is similar to the `date`, but includes a timezone. 🤗

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+CREATE TABLE
+  invoices(
+    id INT GENERATED ALWAYS AS IDENTITY,
+    total_amount VARCHAR(20),
+    generated_at DATE,
+    payed_at DATE,
+    medical_history_id INT REFERENCES medical_histories(id),
+    PRIMARY KEY(id)
+  );
```
Comment (abdulhamiid) [inline schema_based_on_diagram.sql]:
```sql
total_amount VARCHAR(20),
generated_at DATE,
payed_at DATE,
```

According to the database diagram, kindly add the correct `datatype` in the columns above: 🛠🤓

* The datatype parameter specifies the type of data the column can hold (e.g. varchar, decimal, integer, etc.).

Diagram 👀

image



- [ ] Tip: For an overview of the available data types, visit the complete [Data Types Reference](https://www.w3schools.com/sql/sql_datatypes.asp).

File: schema_based_on_diagram.sql [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+CREATE TABLE
+  invoice_items(
+    id INT GENERATED ALWAYS AS IDENTITY,
+    invoice_id INT REFERENCES invoices(id),
+    treatment_id INT REFERENCES treatments(id),
+    PRIMARY KEY(id)
+  );
```
Comment (abdulhamiid) [inline schema_based_on_diagram.sql]: Please, kindly add the missing columns in the table `invoice_items`:

image

```
unit_price
quantity
total_price
```

File: schema_based_on_diagram.sql [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+CREATE TABLE
+  treatments(
+    id INT GENERATED ALWAYS AS IDENTITY,
+    name VARCHAR(120),
+    type VARCHAR(120),
+    PRIMARY KEY(id)
+  );
```
Comment (abdulhamiid) [inline schema_based_on_diagram.sql:15]: - [ ] According to the [project requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/database_from_diagram.md), please lets all FOREIGN KEYS have indexes created:

```
CREATE INDEX ON medical_histories (patient_id);
CREATE INDEX ON invoices (medical_history_id);
CREATE INDEX ON invoice_items (invoice_id);
CREATE INDEX ON invoice_items (treatment_id);
CREATE INDEX ON medical_histories_has_treatments (medical_history_id);
CREATE INDEX ON medical_histories_has_treatments (treatment_id);
```

- [ ] Foreign keys are very useful for keeping a database clean and won't affect the `SELECT` statements. 

- [ ] Indexes help you quickly browse the database and find the data you need. Knowing how and when to use them will improve database performance and make your work easier. 💪

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi Team 👋, 
Alt Text
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Descriptions are descriptive
- ✔️ Interpret a database diagram and implement the SQL needed to build the database.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

[REQUIRED] - Kindly fix the broken links in the `Authors` section, let's avoid broken links in our readme for a professional repository. 

image


Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Kossi-stack — APPROVED** [review]
## Status: Approved ✔️
**Hi @osmanwako,**

You did a really great job! 💯 💯💯💯
Congratulations! 🎉 🎉 🎉 🎉 🎉


You may close this Pull Request and merge the changes
Keep being awesome dev's and good luck with your next milestone!

## To Highlight

✅ Descriptive pull request message
✅ Documented your work professionally
✅ README file is descriptive.
✅ Commit messages are meaningful.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding! 👏 👏 👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**